dvc/dependency/repo.py

```diff
@@ -112,64 +122,63 @@ def get_used_objs(
                 self.def_repo[self.PARAM_REV_LOCK] = rev
 
             path_info = PathInfo(repo.root_dir) / str(self.def_path)
-            try:
-                for odb, objs in repo.used_objs(
-                    [os.fspath(path_info)],
-                    force=True,
-                    jobs=kwargs.get("jobs"),
-                    recursive=True,
-                ).items():
-                    if odb is None:
-                        odb = repo.cloud.get_remote().odb
-                    self._check_circular_import(odb)
-                    used_objs[odb].update(objs)
-            except (NoRemoteError, NoOutputOrStageError):
-                pass
-
+            if not obj_only:
```

Comment:
Could you elaborate on `obj_only`, please? Not sure I get it right away.

Reply:
We used to have a lot of duplicated code in `get_obj` and `get_used_objs`/`get_used_cache` that did this same `_make_repo()` logic in both functions, so I unified them here.

The only difference is that for `get_used_objs` we need to do the full recursive used objects collection for imported outs, but for `get_obj` we can skip that collection (via `obj_only=True`)